rust/operator-binary/src/crd/mod.rs

```diff
@@ -155,25 +173,24 @@ pub mod versioned {
         #[serde(skip_serializing_if = "Option::is_none")]
         pub vector_aggregator_config_map_name: Option<String>,
 
-        /// This field controls which type of Service the Operator creates for this HiveCluster:
-        ///
-        /// * cluster-internal: Use a ClusterIP service
-        ///
-        /// * external-unstable: Use a NodePort service
-        ///
-        /// * external-stable: Use a LoadBalancer service
-        ///
-        /// This is a temporary solution with the goal to keep yaml manifests forward compatible.
-        /// In the future, this setting will control which [ListenerClass](DOCS_BASE_URL_PLACEHOLDER/listener-operator/listenerclass.html)
-        /// will be used to expose the service, and ListenerClass names will stay the same, allowing for a non-breaking change.
-        #[serde(default)]
-        pub listener_class: CurrentlySupportedListenerClasses,
-
         /// Settings related to user [authentication](DOCS_BASE_URL_PLACEHOLDER/usage-guide/security).
         pub authentication: Option<AuthenticationConfig>,
     }
 }
 
+impl Default for v1alpha1::HiveMetastoreRoleConfig {
+    fn default() -> Self {
+        v1alpha1::HiveMetastoreRoleConfig {
+            listener_class: metastore_default_listener_class(),
+            common: Default::default(),
+        }
+    }
+}
+
+fn metastore_default_listener_class() -> String {
+    "cluster-internal".to_string()
+}
```

Comment:
For mine, I used a constant:

```rust
pub const DEFAULT_LISTENER_CLASS: &str = "cluster-internal";
```

and

```suggestion
fn metastore_default_listener_class() -> String {
    DEFAULT_LISTENER_CLASS.to_owned()
}
```

Reply:
done with [8637ecd](https://github.com/stackabletech/hive-operator/pull/605/commits/8637ecdf1f80cb942ef2cf78c75344d9266ae4af)